src/OrleansServiceBus/Providers/Streams/EventHub/EventHubPartitionCheckpoint.cs

```diff
@@ -0,0 +1,74 @@
+﻿
+using System;
+using System.Threading.Tasks;
+using Microsoft.ServiceBus.Messaging;
+using Orleans.AzureUtils;
+
+namespace Orleans.ServiceBus.Providers
+{
+    internal class EventHubPartitionCheckpoint
```

Comment:
This class is making the EventHub stream provider dependent on Azure storage (and Orleans Azure utilities). I think it should be fine until we call this stream provider "stable", but I'd like us to be careful not to abuse this dependency, and in the future be easily extractable (for example, I might want to use local storage or Service Fabric persistence for it).
Can you at least add a comment here about it?